pages/example-6-xss/vulnerable.tsx

```diff
@@ -0,0 +1,40 @@
+import { BlogPost, getSpecialBlogPosts } from '../../database/blogPosts';
+import {CommonContent} from './common';
+
+type Props =
+  | {
+      error: string;
+    }
+  | {
+      blogPosts: BlogPost[];
+    };
+
+export default function MissingAuthenticationGssp(props: Props) {
+  return (
+    <div>
+      <CommonContent />
+
+      {'error' in props && <div style={{ color: 'red' }}>{props.error}</div>}
```

Comment:
maybe it makes sense to move this to the `CommonContent` component, because it's not 

I'll also change Example 4 to use a similar pattern

Reply:
First part done in 547b398